lib/parser/provenance.cc

```diff
@@ -241,6 +252,26 @@ const SourceFile *AllSources::GetSourceFile(
       origin.u);
 }
 
+std::optional<SourcePosition> AllSources::GetSourcePosition(
+    Provenance prov) const {
+  const Origin &origin{MapToOrigin(prov)};
+  if (const auto *inc{std::get_if<Inclusion>(&origin.u)}) {
+    std::size_t offset{origin.covers.MemberOffset(prov)};
+    return SourcePosition{inc->source.FindOffsetLineAndColumn(offset)};
```

Comment:
It seems like `FindOffsetLineAndColumn` should return `SourcePosition` too.

Reply:
True, so maybe it's not the right thing to call here. But given the addition of the `SourcePosition` type, `FindOffsetLineAndColumn()` should be modified to make use of it.